engine/src/main/java/org/destinationsol/SolApplication.java

```diff
@@ -121,57 +123,62 @@
 
     private ComponentManager componentManager;
 
+    private BeanContext appContext;
+    private BeanContext gameContext;
+
     // TODO: Make this non-static.
     private static Set<ResizeSubscriber> resizeSubscribers;
 
-    public SolApplication(ModuleManager moduleManager, float targetFPS) {
+    @Inject
+    protected SolApplication() {
+        throw  new RuntimeException("Unimplemented");
```

Comment:
Maybe make the error message here a bit more helpful? Perhaps something like `Cannot instantiate SolApplication via DI`?